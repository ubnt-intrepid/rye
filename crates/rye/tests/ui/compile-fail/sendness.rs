fn main() {}

#[rye::test(non_send)]
async fn incorrect_arg(cx: &mut rye::Context<'_>) {
    let rc = std::rc::Rc::new(());
    (async {}).await;
    drop(rc);
}

#[rye::test(?Send)]
fn nosend_in_sync_fn(cx: &mut rye::Context<'_>) {}
