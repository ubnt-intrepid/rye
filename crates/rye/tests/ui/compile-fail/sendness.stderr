error: only '?Send' or '!Send' is accepted
 --> $DIR/sendness.rs:3:13
  |
3 | #[rye::test(non_send)]
  |             ^^^^^^^^

error: accepted only for async functions
  --> $DIR/sendness.rs:10:13
   |
10 | #[rye::test(?Send)]
   |             ^^^^^
